certbot/client.py

```diff
@@ -410,11 +410,13 @@ def enhance_config(self, domains, config, chain_path):
             logger.warning("No config is specified.")
             raise errors.Error("No config available")
 
-        supported = self.installer.supported_enhancements()
-        redirect = config.redirect if "redirect" in supported else False
-        hsts = config.hsts if "ensure-http-header" in supported else False
-        uir = config.uir if "ensure-http-header"  in supported else False
-        staple = config.staple if "staple-ocsp" in supported else False
+
+        self._verify_all_config_options_supported(config)
+
+        redirect = getattr(config, "redirect", None)
```

Comment:
Done. Reverted this to what it was before and am doing everything in the constructor now